mission_control/navigator_missions/nav_missions/pinger.py

```diff
@@ -0,0 +1,198 @@
+
+        pose = yield self.navigator.tx_pose()
+        self.dist_sign_pos = np.linalg.norm(pose[:2] - (self.gate_1_pos + self.OBSERVE_DISTANCE_METERS * self.g_perp) )
+        self.dist_sign_neg = np.linalg.norm(pose[:2] - (self.gate_1_pos - self.OBSERVE_DISTANCE_METERS * self.g_perp) )
+        if np.argmin( np.array( [self.dist_sign_pos,self.dist_sign_neg] ) ) == 1:
+            self.negate = True
+        if self.negate:
+            self.observation_points =  ( np.append( (self.gate_1_pos - self.observer_sign * self.OBSERVE_DISTANCE_METERS * self.g_perp), 0),
+                                         np.append( (self.gate_3_pos - self.OBSERVE_DISTANCE_METERS * self.g_perp), 0) )
+        else:
+             self.observation_points =  ( np.append( (self.gate_1_pos + self.observer_sign * self.OBSERVE_DISTANCE_METERS * self.g_perp), 0),
+                                         np.append( (self.gate_3_pos + self.OBSERVE_DISTANCE_METERS * self.g_perp), 0) )
+        self.look_at_points = (np.append(self.gate_1_pos,0), np.append(self.gate_3_pos,0))
+        self.observation_markers()
+
+    @txros.util.cancellableInlineCallbacks
+    def search_samples(self):
+        for i,p in enumerate(self.observation_points):
+          yield self.stop_listen()
+          yield self.navigator.move.set_position(p).look_at(self.look_at_points[i]).go()
+          yield self.start_listen()
+          print "Listening at ",p
+          yield self.navigator.nh.sleep(self.LISTEN_TIME)
+        yield self.stop_listen()
+
+    @txros.util.cancellableInlineCallbacks
+    def get_pinger_pose(self):
+        res = yield self.pinger_client(FindPingerRequest())
+        pinger_pose = res.pinger_position
+        self.pinger_pose = navigator_tools.rosmsg_to_numpy(pinger_pose)[:2]
+        self.distance1 = np.linalg.norm(self.pinger_pose - self.gate_1_pos) 
+        self.distance2 = np.linalg.norm(self.pinger_pose - self.gate_2_pos)
+        self.distance3 = np.linalg.norm(self.pinger_pose - self.gate_3_pos)
+
+    @txros.util.cancellableInlineCallbacks
+    def go_thru_gate(self):
+        self.gate_index = np.argmin( np.array( [self.distance1,self.distance2,self.distance3] ) )
+        self.gate_thru_poses  = self._thru_gate_poses( (self.gate_1_pos,self.gate_2_pos,self.gate_3_pos), self.gate_index)
+        self.gate_thru_markers()
+        for p in self.gate_thru_poses:
+            yield self.navigator.move.set_position(p).go(initial_plan_time=5)
+
+    @txros.util.cancellableInlineCallbacks
+    def set_active_pinger(self):
+      if self.gate_index == 0:
+          yield self.navigator.mission_params["acoustic_pinger_active"].set("RED-WHITE")
+      elif self.gate_index == 1:
+          yield self.navigator.mission_params["acoustic_pinger_active"].set("WHITE-WHITE")
+      elif self.gate_index == 2:
+          yield self.navigator.mission_params["acoustic_pinger_active"].set("WHITE-GREEN")
+
+    def _get_perp(self,g1,g3):
+        assert len(g1) == 2
```

Comment:
PEP in the function call.
Also, I wouldn't throw an assertion here, just do `g1 = np.array([g1])[:2]` which is much more functional (since you could  pass in a non numpy array also).